planet/clients/mosaics.py

```diff
@@ -0,0 +1,397 @@
+        """
+        List the a mosaic's quads.
+
+        Example:
+
+        ```
+        mosaic = await client.get_mosaic("d5098531-aa4f-4ff9-a9d5-74ad4a6301e5")
+        quads = await client.list_quads(mosaic)
+        async for q in quads:
+            print(q)
+        ```
+        """
+        if geometry:
+            resp = await self._quads_geometry(mosaic,
+                                              geometry,
+                                              minimal,
+                                              summary)
+        else:
+            if bbox is None:
+                xmin, ymin, xmax, ymax = cast(BBox, mosaic['bbox'])
+                search = (max(-180, xmin),
+                          max(-85, ymin),
+                          min(180, xmax),
+                          min(85, ymax))
+            else:
+                search = bbox
+            resp = await self._quads_bbox(mosaic, search, minimal, summary)
+        # kinda yucky - yields a different "shaped" dict
+        if summary:
+            yield resp.json()["summary"]
+            return
+        async for item in MosaicQuads(resp, self._session.request):
+            yield item
+
+    async def _quads_geometry(self,
+                              mosaic: dict,
+                              geometry: dict,
+                              minimal: bool,
+                              summary: bool) -> Response:
+        params = {}
+        if minimal:
+            params["minimal"] = "true"
+        if summary:
+            params["summary"] = "true"
+        mosaic_id = mosaic["id"]
+        return await self._session.request(
+            method="POST",
+            url=self._url(f"mosaics/{mosaic_id}/quads/search"),
+            params=params,
+            json=geometry,
+        )
+
+    async def _quads_bbox(self,
+                          mosaic: dict,
+                          bbox: BBox,
+                          minimal: bool,
+                          summary: bool) -> Response:
+        quads_template = mosaic["_links"]["quads"]
+        # this is fully qualified URL, so don't use self._url
+        url = quads_template.replace("{lx},{ly},{ux},{uy}",
+                                     ",".join([str(f) for f in bbox]))
+        # params will overwrite the templated query
+        if minimal:
+            url += "&minimal=true"
+        if summary:
+            url += "&summary=true"
+        return await self._session.request(
+            method="GET",
+            url=url,
+        )
+
+    async def get_quad(self, name_or_id: str, quad_id: str) -> dict:
+        """
+        Get a mosaic's quad information.
+
+        Example:
+
+        ```
+        quad = await client.get_quad("d5098531-aa4f-4ff9-a9d5-74ad4a6301e5", "1234-5678")
+        print(quad)
+        ```
+        """
+        if not _is_uuid(name_or_id):
+            mosaic = await self.get_mosaic(name_or_id)
+            if mosaic is None:
+                return {}
```

Comment:
Along the same lines as the comment about the `Optional[dict]` return value, raising some kind of MosaicNotFoundException here would be more clear.

Reply:
also I notice some methods require the metadata dict up front, and others (like this one) fetch it under the hood by name/ID. It would be great to standardize on accepting either, fetching metadata if needed.